lib/procore-sift.rb

```diff
@@ -39,20 +39,28 @@ def filter_errors
   def filter_validator
     @_filter_validator ||= FilterValidator.build(
       filters: filters,
-      sort_fields: self.class.sort_fields,
+      sort_fields: sort_fields,
       filter_params: filter_params,
       sort_params: sort_params,
     )
   end
 
   def filters
-    self.class.filters
+    self.class.ancestors
+      .take_while { |klass| klass.name != "Sift" }
+      .flat_map { |klass| klass.try(:filters) }
+      .compact
+      .uniq { |f| [f.param, f.class] }
   end
 
   def sorts_exist?
     filters.any? { |filter| filter.is_a?(Sort) }
   end
 
+  def sort_fields
+    self.class.ancestors.flat_map { |klass| klass.try(:sort_fields) }.compact
```

Comment:
Should we add the same `take_while` call here?

Reply:
Oh right, yup!